WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -1087,6 +1087,19 @@ extension StatsInsightsStore {
         }
     }
 
+    var followersInsightsDetailsStatus: StoreFetchingStatus {
+        switch (followersTotalsStatus, state.dotComFollowersStatus, state.emailFollowersStatus) {
+        case (let a, let b, let c) where a == .loading || b == .loading || c == .loading:
+            return .loading
+        case (let a, let b, let c) where a == .error || b == .error || c == .error:
+            return .error
+        case (.success, .success, .success):
+            return .success
+        default:
+            return .idle
+        }
```

Comment:
@sla8c, the `a`, `b`, and `c` temporary variables and advanced switch statement make this a bit hard for me to follow. Would something like this work and be simpler? 

```suggestion
        let statuses = [followersTotalsStatus, state.dotComFollowersStatus, state.emailFollowersStatus]
        if statuses.contains(.loading) {
            return .loading
        } else if statuses.contains(.error) {
            return .error
        } else if statuses.allSatisfy({ $0 == .success }) {
            return .success
        } else {
            return .idle
        }
```

Reply:
@Guarani @staskus looking at this with a fresh set of eyes, I have removed my function here however I added a unit test to test the existing function but did not refactor it yet (I'll do so in another PR)

Reply:
Could you expand a bit more about what @guarani asked before: the differences between 2 methods and do you know why exactly the issue happens?

Reply:
@staskus the difference was that the new function I added was redundant and I could just re-use `followersTotalsStatus`